cluster.py

```diff
@@ -1,21 +1,46 @@
 import logging
+import socket
+import time
 from pathlib import Path
-from typing import Dict
+from typing import Dict, Tuple
 
 from ccmlib import scylla_cluster as ccm
 
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
 
+def get_cluster_id() -> Tuple[socket.socket, int]:
+    """gets integer number that will be used for ip prefix of cluster nodes,
+    so it's possible to run tests in parallel.
+
+    Returns tuple of socket and cluster id, where socket later needs to be closed."""
+    logger.info("Getting cluster id for support parallel tests...")
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    for cluster_id in range(126):
+        try:
+            sock.bind((f'127.0.{cluster_id}.1', 48783))  # random port
+            logger.info("Cluster id acquired: %s", cluster_id)
+            return sock, cluster_id
+        except OSError:
+            continue
+    raise ValueError(f"Couldn't acquire cluster id - looks clusters are not cleared properly")
+
+
+def release_cluster_lock(sock: socket.socket) -> None:
+    sock.close()
+
+
 class TestCluster:
     """Responsible for configuring, starting and stopping cluster for tests"""
 
     def __init__(self, driver_directory: Path, version: str, configuration: Dict[str, str]) -> None:
         self.cluster_directory = driver_directory / "ccm"
         self.cluster_directory.mkdir(parents=True, exist_ok=True)
         logger.info("Preparing test cluster binaries and configuration...")
+        self._cluster_lock, id = get_cluster_id()
```

Comment:
`id` is build-in python function. It is not recommended to overwrite it.
Then, can you please use `self._cluster_sock` instead of the `self._cluster_lock`?
It will reduce the confusion, IMHO.

Reply:
agreed to `id`, I forgot about it.
as to `sock` - socket is implementation detail of machine-wide lock. Implementation has no much meaning, that's why I try to hide it behind 'lock'.
I did some renaming in that area, please check if it fits better.